protobuf/types/struct.proto

```diff
@@ -0,0 +1,115 @@
+// Protocol Buffers - Google's data interchange format
+// Copyright 2008 Google Inc.  All rights reserved.
+// https://developers.google.com/protocol-buffers/
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+//     * Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above
+// copyright notice, this list of conditions and the following disclaimer
+// in the documentation and/or other materials provided with the
+// distribution.
+//     * Neither the name of Google Inc. nor the names of its
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+// OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+syntax = "proto3";
+
+import "gogo/protobuf/gogoproto/gogo.proto";
+
+package mesg.protobuf;
```

Comment:
why not just `type` like the rest of our messages? This would be easier to generate for the engine but also mesg-js lib

Reply:
> why not just type like the rest of our messages? 

I wanted to put emphasis that is still google.protobuf.Struct and not some of our types like event, execution etc...

> This would be easier to generate for the engine but also mesg-js lib

How is this gonna make it easier? Before was google.protobuf and did you find trouble with that?

I can put any type here, but the mesg.protobuf package name was only dictated to not include this "sepcial" case in all our types.

Reply:
Ok i'm fine to keep it like that, I agree it's nice to have something "special".

I just have a hard time to try to generate the types for mesg-js from this script 
https://github.com/mesg-foundation/mesg-js/blob/master/copy-proto and we also need to update the doc here https://github.com/mesg-foundation/docs/blob/master/generate-docs-api.sh

Maybe if you can have a look that would be great, I might be struggling on something for no reasons

Reply:
```bash
#!/bin/bash -e

DEF_REPO="mesg-foundation/core"
DEF_BRANCH="${1:-dev}"
PROTO_PATH="https://raw.githubusercontent.com/$DEF_REPO/$DEF_BRANCH/protobuf"
DIR="./"

rm -rf src/protobuf
rm -rf src/api/typedef
mkdir -p src/protobuf/api
mkdir -p src/protobuf/types
mkdir -p src/api/typedef

curl -so "./src/protobuf/types/struct.proto" "$PROTO_PATH/types/struct.proto"

ressources="event execution instance service process"
for ressource in $ressources
do
  curl -so "./src/protobuf/api/${ressource}.proto" "$PROTO_PATH/api/${ressource}.proto"
  curl -so "./src/protobuf/types/${ressource}.proto" "$PROTO_PATH/types/${ressource}.proto"
  npx pbjs \
    -t static-module \
    --no-create \
    --no-encode \
    --no-decode \
    --no-verify \
    --no-convert \
    --no-delimited \
    --path ./src \
    -o "./src/api/typedef/$ressource.js" \
    "./src/protobuf/types/${ressource}.proto" \
    "./src/protobuf/api/${ressource}.proto"

  npx pbts --path ./src -o "./src/api/typedef/$ressource.d.ts" --name mesg "./src/api/typedef/$ressource.js"
  rm "./src/api/typedef/$ressource.js"
done

curl -so "./src/protobuf/api/core.proto" "$PROTO_PATH/coreapi/api.proto"
```

changes:
- (nice) sh to bash + added -e
- (nice) add -s option to curl to make output cleaner
- **(must)** get the struct.proto first
- **(must)** add --path - prevoius version worked because struct.proto was in /usr/local/include (or similar on mac) which is listed by protoc. Here the location of `struct.proto` has changed so I've added `--path` option to tell pbjs to search that directory.

It's already late so I'm leaving commit to you. The script works:

```
➜  mesg-js git:(master) ✗ gst
On branch master
Your branch is up to date with 'origin/master'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   copy-proto
	modified:   src/api/typedef/event.d.ts
	modified:   src/api/typedef/execution.d.ts
	modified:   src/api/typedef/instance.d.ts
	modified:   src/api/typedef/process.d.ts
	modified:   src/api/typedef/service.d.ts
	modified:   src/protobuf/api/core.proto
	modified:   src/protobuf/api/event.proto
	modified:   src/protobuf/api/execution.proto
	modified:   src/protobuf/api/instance.proto
	modified:   src/protobuf/api/service.proto
	modified:   src/protobuf/types/event.proto
	modified:   src/protobuf/types/execution.proto

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	src/protobuf/types/struct.proto
```

NOTE:

- you can add `.sh` extension to script name :)
- DIR variable is not used
- DEF_REPO is used once and can be removed
- rm & mkdir can be squashed
- Script download coreapi/api.proto but dosen't generate typedef, the question is - is it really needed to download it then?

script after all above modifications:

```bash
#!/bin/bash -e

DEF_BRANCH="${1:-dev}"
PROTO_PATH="https://raw.githubusercontent.com/mesg-foundation/core/$DEF_BRANCH/protobuf"

rm -rf src/protobuf src/api/typedef
mkdir -p src/protobuf/api src/protobuf/types src/api/typedef

curl -so "./src/protobuf/types/struct.proto" "$PROTO_PATH/types/struct.proto"

for ressource in event execution instance service process
do
  curl -so "./src/protobuf/api/${ressource}.proto" "$PROTO_PATH/api/${ressource}.proto"
  curl -so "./src/protobuf/types/${ressource}.proto" "$PROTO_PATH/types/${ressource}.proto"

  npx pbjs \
    -t static-module \
    --no-create \
    --no-encode \
    --no-decode \
    --no-verify \
    --no-convert \
    --no-delimited \
    --path ./src \
    -o "./src/api/typedef/$ressource.js" \
    "./src/protobuf/types/${ressource}.proto" \
    "./src/protobuf/api/${ressource}.proto"

  npx pbts --path ./src -o "./src/api/typedef/$ressource.d.ts" --name mesg "./src/api/typedef/$ressource.js"
  rm "./src/api/typedef/$ressource.js"
done
```

Reply:
Pull request created on mesg-js, thanks @krhubert for that https://github.com/mesg-foundation/mesg-js/pull/130
Also did some testing on the CLI and this seems to work correctly